### PR TITLE
MGMT-4673 Sort ValidationsOutput to prevent uneeded updates

### PR DESCRIPTION
**Pull request by oshercc**

### CODE REVIEW

File: internal/cluster/refresh_status_preprocessor.go [hunk 1]
```diff
@@ -89,7 +90,23 @@ func (r *refreshPreprocessor) preprocess(ctx context.Context, c *clusterPreproce
 	for _, condition := range r.conditions {
 		stateMachineInput[condition.id.String()] = condition.fn(c)
 	}
-	return stateMachineInput, validationsOutput, nil
+	sortedValidationsOutput := sortValidationsOutput(validationsOutput)
+
+	return stateMachineInput, sortedValidationsOutput, nil
+}
+
+func sortValidationsOutput(validationsOutput map[string][]ValidationResult) map[string][]ValidationResult {
```
Comment (filanov) [inline internal/cluster/refresh_status_preprocessor.go:98]: You are sorting the categories and not the results, the problem is with `ValidationResult` the results.

Comment (pkliczewski) [reply internal/cluster/refresh_status_preprocessor.go:98]: Please add UT as well

File: internal/cluster/transition_test.go [hunk 2]
```diff
@@ -3970,6 +3970,22 @@ var _ = Describe("Single node", func() {
 	})
 })
 
+var _ = Describe("ValidationResult sort", func() {
```
Comment (filanov) [inline internal/cluster/transition_test.go:3973]: Need to check the flow with `preprocess`

**Comment by gamli75** [comment]
we have the same issue in the host validations - see https://github.com/openshift/assisted-service/blob/master/internal/host/refresh_status_preprocessor.go#L55 - please fix it there also

File: internal/cluster/transition_test.go (hunk 2 above)
Comment (gamli75) [reply internal/cluster/transition_test.go:3973]: you can do it here https://github.com/openshift/assisted-service/blob/master/internal/cluster/cluster_test.go#L2028

File: internal/cluster/refresh_status_preprocessor.go [hunk 3]
```diff
@@ -89,9 +90,19 @@ func (r *refreshPreprocessor) preprocess(ctx context.Context, c *clusterPreproce
 	for _, condition := range r.conditions {
 		stateMachineInput[condition.id.String()] = condition.fn(c)
 	}
+	for _, validationResults := range validationsOutput {
+		sortByValidationResultID(validationResults)
+	}
 	return stateMachineInput, validationsOutput, nil
 }
 
+func sortByValidationResultID(validationResults []ValidationResult) {
```
Comment (pkliczewski) [inline internal/cluster/refresh_status_preprocessor.go:99]: What do you think about moving it to utils. We use it in 2 places.

Comment (flaper87) [reply internal/cluster/refresh_status_preprocessor.go:99]: Was literally thinking this while doing the review! It'd be nice to do it

Comment (pkliczewski) [reply internal/cluster/refresh_status_preprocessor.go:99]: Even though `ValidationResult` looks similar is different type. There is different `validationID` type used for host is `models.HostValidationID` and for the cluster is `models.ClusterValidationID`. We can't easily unify those types.

**Comment by pkliczewski** [comment]
I rebased this PR and reposted it here: https://github.com/openshift/assisted-service/pull/1373

**Comment by gamli75** [comment]
was replaced by https://github.com/openshift/assisted-service/pull/1373

**Pull request closed without merging**